src/Serializer/SnippetDenormalizer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace eLife\ApiSdk\Serializer;
+
+use eLife\ApiSdk\Promise\CallbackPromise;
+use GuzzleHttp\Promise\PromiseInterface;
+
+final class SnippetDenormalizer
+{
+    private $id;
+    private $fetcher;
+    private $identityMap;
+    private $globalCallback;
+
```

Comment:
```
/**
 * @param callable $id  accepts an array representing ... returns a string representing ...
 * @param callable $fetcher  ...
 */
```
probably some unit tests for documentation will be helpful in understanding the responsibility of this class.